src/libs/libpng/lv_libpng.c

```diff
@@ -303,6 +303,7 @@ static lv_draw_buf_t * decode_png(lv_image_decoder_dsc_t * dsc)
         if(dsc->src_type == LV_IMAGE_SRC_FILE) {
             LV_LOG_ERROR("alloc PNG_IMAGE_SIZE(%" LV_PRIu32 ") failed: %s", (uint32_t)PNG_IMAGE_SIZE(image),
                          (const char *)dsc->src);
+            png_image_free(&image);
```

Comment:
Should we move logic to just before return NULL:
```
png_image_free(&image);
return NULL;
```

Reply:
Currently in the error handling branch, no need to return NULL in advance.

Reply:
I think @XuNeo means we should handle this case too:
![image](https://github.com/user-attachments/assets/1217ad35-8880-49ac-b466-a45d2f8ca0ff)

which makes sense to me.

BTW, where `png_data` is freed?

Reply:
`png_data`  is free'd in line 306, which is only necessary for file type of image.